app/src/main/java/com/arctouch/movies/movie/dao/MovieLocalDAO.java

```diff
@@ -0,0 +1,79 @@
+package com.arctouch.movies.movie.dao;
+
+import com.arctouch.movies.db.AppDatabaseTransaction;
+import com.arctouch.movies.movie.dao.entity.MovieEntity;
+import com.arctouch.movies.movie.model.Movie;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmResults;
+
+/**
+ * Created by marcelo.alves on 06/09/17.
+ */
+
+public class MovieLocalDAO extends AppDatabaseTransaction<Movie> {
```

Comment:
Isn't it dangerous have all methods get and close new instances of Realm while using the same variable? I don't think this is thread-safe, is it?

Reply:
For this project version it is! But this is not the better way and not safe ! really need an improve!